File: Service/LivestreamApp.Server/Shutdown/ShutdownSimulationService.cs
using Ninject.Extensions.Logging;
using System;

namespace LivestreamApp.Server.Shutdown
{
    public class ShutdownSimulationService : IShutdownService
    {
        private readonly ILogger _logger;

        public ShutdownSimulationService(ILogger logger)
        {
            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
        }

        public void ShutdownServer()
        {
            _logger.Info("Simulating shutting down server - shutdown will not be executed.");
        }

        public void RestartServer()
        {
            _logger.Info("Simulating restarting server - restart will not be executed.");
        }
    }
}

File: Service/src/core/system/shutdown/shutdown-simulation-service.ts
import { injectable } from "inversify";
import { Logger } from "../../util/logger";

/**
 * Implementation for shutdown simulation
 */
@injectable()
export class ShutdownSimulationService {

    constructor(private logger: Logger) { }

    /** @inheritdoc */
    public shutdown(): void {
        this.logger.info("Simulating shutdown.");
    }
}
